internal/service/streamprivatelinkendpoint/data_source.go

```diff
@@ -37,19 +36,20 @@ func (d *ds) Read(ctx context.Context, req datasource.ReadRequest, resp *datasou
 		return
 	}
 
-	// TODO: make get request to resource
-
-	// connV2 := d.Client.AtlasV2
-	//if err != nil {
-	//	resp.Diagnostics.AddError("error fetching resource", err.Error())
-	//	return
-	//}
-
-	// TODO: process response into new terraform state
-	// newStreamPrivatelinkEndpointModel, diags := NewTFModel(ctx, apiResp)
-	// if diags.HasError() {
-	// 	resp.Diagnostics.Append(diags...)
-	// 	return
-	// }
-	// resp.Diagnostics.Append(resp.State.Set(ctx, newStreamPrivatelinkEndpointModel)...)
+	projectID := tfModel.ProjectId.ValueString()
+	connectionID := tfModel.Id.ValueString()
+
+	connV2 := d.Client.AtlasV2
+	streamsPrivateLinkConnection, _, err := connV2.StreamsApi.GetPrivateLinkConnection(ctx, projectID, connectionID).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("error fetching resource", err.Error())
+		return
+	}
+
+	newStreamPrivatelinkEndpointModel, diags := NewTFModel(ctx, projectID, streamsPrivateLinkConnection)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newStreamPrivatelinkEndpointModel)...)
```

Comment:
similar to @EspenAlbert idea, if you use diags as input param, it's a bit more straightforward and you don't need to assign it back, e.g.: 

```suggestion
	projectID := tfModel.ProjectId.ValueString()
	connectionID := tfModel.Id.ValueString()
      diags := &resp.Diagnostics
	connV2 := d.Client.AtlasV2
	streamsPrivateLinkConnection, _, err := connV2.StreamsApi.GetPrivateLinkConnection(ctx, projectID, connectionID).Execute()
	if err != nil {
		diags.AddError("error fetching resource", err.Error())
		return
	}

	newStreamPrivatelinkEndpointModel := NewTFModel(ctx, diags, projectID, streamsPrivateLinkConnection)
	if diags.HasError() {
		return
	}
     diags.Append(resp.State.Set(ctx, newStreamPrivatelinkEndpointModel)...)
```